### ISSUE TITLE
Question: Selected Part of elmo embedding

### ISSUE THREAD

**Issue by jeniyat** [allenai/allennlp#1995]
#Is it possible to use the character embedding only(I want to exclude word embedding and context embedding) for NER task?

**Comment by nelson-liu** [comment]
I'm a little bit confused by what you want---do you just want to use the output of the pretrained elmo character convnet as your representation?

If so, you can add the key `"scalar_mix_parameters": [1.0, 0.0, 0.0]` to the elmo specification in the config, and the representations returned by elmo will just be the character-convnet-generated vectors.

Closing this for now, feel free to reopen if that doesn't answer your question.

**Comment by matt-peters** [comment]
Yes, you can accomplish this though the method that Nelson mentioned. However, note that we apply a softmax to the `scalar_mix_parameters` so you need to specify parameters as e.g. `"scalar_mix_parameters": [50.0, 0.0, 0.0]` to only select the first layer.

**Comment by lecholin** [comment]
> Yes, you can accomplish this though the method that Nelson mentioned. However, note that we apply a softmax to the `scalar_mix_parameters` so you need to specify parameters as e.g. `"scalar_mix_parameters": [50.0, 0.0, 0.0]` to only select the first layer.

May I ask how to select the FIRST and the SECOND layer? Do I have to define two Elmo class with scalar_mix_parameters=[50.0, 0.0, 0.0] and [0.0, 50.0, 0.0] respectively?

**Comment by matt-gardner** [comment]
What do you want to do with those two layers?  Average them?  Or use them separately for different things?  If you want to average them, then `[50.0, 50.0, 0.0]` would work.  If you want to use them separately, you could use two `Elmo` classes, or you could set `num_output_representations=2`, though it looks like we can't currently configure different scalar mixes for multiple output representations...

Alternatively, depending on your goals, something like @nelson-liu's [contextual analysis repo](https://github.com/nelson-liu/contextual-repr-analysis) might be a good fit.

**Comment by yanaiela** [comment]
Opening this again as I also faced this issue, and there might be a simple solution.
In https://github.com/allenai/allennlp/blob/master/allennlp/modules/token_embedders/elmo_token_embedder.py#L62 there's a hard coded parameter which sets the number of resulting layers, currently it's set to 1. Can't we simply add a customized parameter? It can be set to 1 as default to not break things, but then users will have control on how many layers to get back from elmo.
